experimental/PropertyDDS/packages/property-dds/src/propertyTree.ts

```diff
@@ -157,25 +164,24 @@ export class SharedPropertyTree extends SharedObject {
 		this._root.cleanDirty(BaseProperty.MODIFIED_STATE_FLAGS.DIRTY);
 	}
 
-	public get changeSet() {
-		// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+	public get activeCommit(): CommitNode {
 		return this.tipView;
 	}
 	public get root(): NodeProperty {
 		return this._root as NodeProperty;
 	}
 
-	public commit(submitEmptyChange = false) {
+	public commit(submitEmptyChange = false, metadata: Metadata = {}) {
 		const changes = this._root._serialize(true, false, BaseProperty.MODIFIED_STATE_FLAGS.PENDING_CHANGE);
         if (submitEmptyChange || !_.isEmpty(changes)) {
```

Comment:
a question that is open here, should the metadata override the submission flag e.g should it commit an empty changeset if metadata is provieded?

Reply:
Interesting question. 

It is certainly a valid scenario to submit a commit with metadata and have an empty changeset. The meta-data could still contain valuable information. And if the client has a custom conflict resolution handler, it could even happen that such a commit will have an effect after rebasing, even if it didn't have an effect when it was created.

On the other hand, we already have a flag, where the user can decide whether to create such a commit or not and there might be cases where you explicitly don't want to create an empty commit, even with meta data.

So I would suggest, we respect the flag, but we change the default behaviour in the sense that if the flag is not explicitly specified and metadata is provided we always create a commit. Admittedly, this increases the complexity of the API a bit, but in my mind that would be the sane default for commits with meta-data.

@karlbom What is you oppinion?

Reply:
I think to allow users more or less passing things in optionally is okay it makes the logic inside somewhat more verbose but for the user they still have the option to completely ignore it if they dont need it.